Mathlib/Combinatorics/SimpleGraph/Bipartite.lean

```diff
@@ -284,4 +285,91 @@ theorem isBipartite_iff_exists_isBipartiteWith :
+
+namespace CompleteBipartiteSubgraph
+
+/-- The "left" and "right" parts in a `G.CompleteBipartiteSubgraph s t` are disjoint. -/
+theorem disjoint_left_right : Disjoint B.left B.right :=
+  disjoint_left.mpr fun v hv₁ hv₂ ↦ (G.loopless v) (B.Adj hv₁ hv₂)
+
+/-- The finset of vertices in a `G.CompleteBipartiteSubgraph s t`. -/
+@[simp]
+abbrev verts : Finset V := disjUnion B.left B.right B.disjoint_left_right
+
+/-- There are `s + t` vertices in a complete bipartite subgraph with `s` vertices in the "left"
+part and `t` vertices in the "right" part. -/
+theorem card_verts : #B.verts = s + t := by simp [card_left, card_right]
+
+/-- A complete bipartite subgraph gives rise to a copy of a complete bipartite graph. -/
+noncomputable def toCopy : Copy (completeBipartiteGraph (Fin s) (Fin t)) G := by
+  have : Nonempty (Fin s ↪ B.left) := by
+    apply Function.Embedding.nonempty_of_card_le
+    rw [Fintype.card_fin, card_coe, card_left]
+  let fs : Fin s ↪ B.left := Classical.arbitrary (Fin s ↪ B.left)
+  have : Nonempty (Fin t ↪ B.right) := by
+    apply Function.Embedding.nonempty_of_card_le
+    rw [Fintype.card_fin, card_coe, card_right]
+  let ft : Fin t ↪ B.right := Classical.arbitrary (Fin t ↪ B.right)
+  let f : Fin s ⊕ Fin t ↪ V := by
+    refine ⟨Sum.elim (Subtype.val ∘ fs) (Subtype.val ∘ ft), fun st₁ st₂ ↦ ?_⟩
+    match st₁, st₂ with
+    | Sum.inl s₁, Sum.inl s₂ => simp [← Subtype.ext_iff]
+    | Sum.inr t₁, Sum.inl s₂ =>
+      simpa using (B.Adj (fs s₂).prop (ft t₁).prop).ne'
+    | Sum.inl s₁, Sum.inr t₂ =>
+      simpa using (B.Adj (fs s₁).prop (ft t₂).prop).symm.ne'
+    | Sum.inr t₁, Sum.inr t₂ => simp [← Subtype.ext_iff]
+  refine ⟨⟨f.toFun, fun {st₁ st₂} hadj ↦ ?_⟩, f.injective⟩
+  rcases hadj with ⟨hst₁, hst₂⟩ | ⟨hst₁, hst₂⟩
+  all_goals dsimp [f]
+  · rw [← Sum.inl_getLeft st₁ hst₁, ← Sum.inr_getRight st₂ hst₂,
+      Sum.elim_inl, Sum.elim_inr]
+    exact B.Adj (by simp) (by simp)
+  · rw [← Sum.inr_getRight st₁ hst₁, ← Sum.inl_getLeft st₂ hst₂,
+      Sum.elim_inl, Sum.elim_inr, adj_comm]
+    exact B.Adj (by simp) (by simp)
+
+/-- A copy of a complete bipartite graph identifies a complete bipartite subgraph. -/
+def ofCopy (f : Copy (completeBipartiteGraph α β) G) :
+    G.CompleteBipartiteSubgraph (card α) (card β) where
+  left := univ.map ⟨f ∘ Sum.inl, f.injective.comp Sum.inl_injective⟩
+  card_left := by rw [card_map, card_univ]
+  right := univ.map ⟨f ∘ Sum.inr, f.injective.comp Sum.inr_injective⟩
+  card_right := by rw [card_map, card_univ]
+  Adj _ h₁ _ h₂ := by
+    rw [mem_map] at h₁ h₂
+    obtain ⟨_, _, h₁⟩ := h₁
+    obtain ⟨_, _, h₂⟩ := h₂
+    rw [← h₁, ← h₂]
+    exact f.toHom.map_adj (by simp)
+
+end CompleteBipartiteSubgraph
+
+/-- Simple graphs contain a copy of a `completeBipartiteGraph α β` iff
+`G.CompleteBipartiteSubgraph (card α) (card β)` is nonempty. -/
+theorem completeBipartiteGraph_isContained_iff :
+    completeBipartiteGraph α β ⊑ G ↔ Nonempty (G.CompleteBipartiteSubgraph (card α) (card β)) :=
```

Comment:
... this would become
```suggestion
theorem completeBipartiteGraph_isContained_iff :
    completeBipartiteGraph α β ⊑ G ↔
      ∃ C : G.CompleteBipartiteSubgraph, C.left.ncard = card α ∧ C.right.ncard = card β :=
```